couch/tests/test_couchv2.py

```diff
@@ -214,81 +216,51 @@ def test_check_without_names(aggregator, gauges):
 
 @pytest.mark.usefixtures('dd_environment')
 @pytest.mark.integration
-def test_only_max_nodes_are_scanned(aggregator, gauges):
+@pytest.mark.parametrize('number_nodes', [1, 2, 3])
+def test_only_max_nodes_are_scanned(aggregator, gauges, number_nodes):
     config = deepcopy(common.NODE1)
     config.pop("name")
-    config['max_nodes_per_check'] = 2
+    config['max_nodes_per_check'] = number_nodes
 
     check = CouchDb(common.CHECK_NAME, {}, [config])
     check.check(config)
 
-    for gauge in gauges["erlang_gauges"]:
-        aggregator.assert_metric(gauge)
-
-    for gauge in gauges["replication_tasks_gauges"]:
-        aggregator.assert_metric(gauge)
-
-    for config in [common.NODE1, common.NODE2]:
-        expected_tags = ["instance:{}".format(config["name"])]
-        for gauge in gauges["cluster_gauges"]:
-            aggregator.assert_metric(gauge, tags=expected_tags)
-
-    for db in ["_users", "_global_changes", "_replicator"]:
-        expected_tags = ["db:{}".format(db)]
-        for gauge in gauges["by_db_gauges"]:
-            aggregator.assert_metric(gauge, tags=expected_tags)
-
-    for db, dd in {"_replicator": "_replicator", "_users": "_auth"}.items():
-        expected_tags = ["design_document:{}".format(dd), "language:javascript", "db:{}".format(db)]
-        for gauge in gauges["by_dd_gauges"]:
-            aggregator.assert_metric(gauge, tags=expected_tags)
-
-    expected_tags = ["instance:{}".format(config["server"])]
-    # One for the version as we don't have any names to begin with
-    aggregator.assert_service_check(CouchDb.SERVICE_CHECK_NAME, status=CouchDb.OK, tags=expected_tags, count=1)
-
-    for node in [common.NODE1, common.NODE2]:
-        expected_tags = ["instance:{}".format(node["name"])]
-        # One for the server stats, the version is already loaded
-        aggregator.assert_service_check(CouchDb.SERVICE_CHECK_NAME, status=CouchDb.OK, tags=expected_tags, count=1)
-
-    expected_tags = ["instance:{}".format(common.NODE3["name"])]
-    for gauge in gauges["cluster_gauges"]:
-        aggregator.assert_metric(gauge, tags=expected_tags, count=0)
```

Comment:
Can you explain what happend to all those asserts? I'm not sure we continue to assert all those things. Do we lose visibility in our tests because of that?

Reply:
Oh, forgot to copy my comments from the Sandbox PR. Here it is:


Note: I simplified some tests to only assert what's necessary:

- `test_only_max_nodes_are_scanned`: assert that that only tags with a limited number of nodes are submitted
- `test_only_max_dbs_are_scanned`:  assert that that only tags with a limited number of dbs are submitted

The removed assertions are already covered in other tests. Example:

https://github.com/DataDog/integrations-core/blob/f70c3e1a5c0a2ecf5d975c3a7baa4fb5661f3097/couch/tests/test_couchv2.py#L84-L119